### PR TITLE
add suffix limit to avoid possible start problems

### PR DESCRIPTION
**Pull request by honganan**
【Improve code robustness】
Here reads all files in the directory with Yaml util, in some situation if the directory contains hidden system file such as `.DS_Store` on Mac OS, and I waste many time on it as the error cannot figure out any clues.

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
It is fine to add this, please update changes.md in the root.

**Comment by wu-sheng** [comment]
And this doc should be updated, https://skywalking.apache.org/docs/main/latest/en/ui/readme/#custom-dashboard. Because you add new requirements for the file name.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModuleProvider.java [hunk 1]
```diff
@@ -376,6 +376,9 @@ public void notifyAfterCompleted() throws ModuleStartException {
         try {
             final File[] templateFiles = ResourceUtils.getPathFiles("ui-initialized-templates");
             for (final File templateFile : templateFiles) {
+                if (!templateFile.getName().endsWith(".yml")) {
```
Comment (kezhenxu94) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModuleProvider.java:379]: Please also take `yaml` into consideration

Comment (honganan) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModuleProvider.java:379]: ok

**Review by wu-sheng — APPROVED** [review]

**Review by kezhenxu94 — APPROVED** [review]
Thank you

**Pull request merged**